File: frontend/src/lib/workflow.ts
import { useQuery } from "@tanstack/react-query";
import { apiFetch } from "./api";

export interface WorkflowNode {
  id: string;
  label: string;
  kind: "start" | "agent" | "approval" | "terminal" | "other";
  x: number;
  y: number;
}

export interface WorkflowEdge {
  id: string;
  source: string;
  target: string;
  trigger: string;
}

export interface WorkflowResponse {
  nodes: WorkflowNode[];
  edges: WorkflowEdge[];
}

export interface WorkflowLiveCount {
  state: string;
  count: number;
}

export interface WorkflowLiveResponse {
  total_tickets: number;
  counts: WorkflowLiveCount[];
}

export function useWorkflow() {
  return useQuery({
    queryKey: ["workflow"] as const,
    queryFn: () => apiFetch<WorkflowResponse>("/api/workflow"),
  });
}

export function useWorkflowLive() {
  return useQuery({
    queryKey: ["workflow", "live"] as const,
    queryFn: () => apiFetch<WorkflowLiveResponse>("/api/workflow/live"),
    refetchInterval: 5_000,
  });
}

File: backend/src/task_summoner/api/schemas/workflow.py
"""Workflow designer response schemas — FSM nodes/edges + live counts."""

from __future__ import annotations

from pydantic import BaseModel


class WorkflowNode(BaseModel):
    id: str
    label: str
    kind: str  # start | agent | approval | terminal
    x: float
    y: float


class WorkflowEdge(BaseModel):
    id: str
    source: str
    target: str
    trigger: str


class WorkflowResponse(BaseModel):
    nodes: list[WorkflowNode]
    edges: list[WorkflowEdge]


class WorkflowLiveCount(BaseModel):
    state: str
    count: int


class WorkflowLiveResponse(BaseModel):
    total_tickets: int
    counts: list[WorkflowLiveCount]


__all__ = [
    "WorkflowEdge",
    "WorkflowLiveCount",
    "WorkflowLiveResponse",
    "WorkflowNode",
    "WorkflowResponse",
]
